### DISCUSSION TITLE
How to install the "M1" version.

### DISCUSSION THREAD

**Discussion by nutmix** [Homebrew/discussions#971] [Getting started]
Hi, I have read a lot of conflicting posts with conflicting ways to install homebrew on M1.

Some say that 

/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

works, others say it doesnt.

Some say you have to clone a rosetta terminal, which is far to arcane.

Some say there is a M1 version of homebrew, but I cant find any mention of it. 

Some recommend not using the M1 version, some recommend not using the non M1 version.

Whats the easiest way to install homebrew on a M1 without rosetta terminal cloning?

**Comment by SMillerDev** [comment]
The easiest way is to run the install command on [brew.sh](https://brew.sh). This will install an ARM native Homebrew.

If you need software that isn't supported on ARM you can also run the install script under Rosetta to install an x86 copy of Homebrew.

**Reply by jamiecropley** [reply]
I literally installed it as it indicates on the homepage: `/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"` I have hardly had any issues, for such a small piece of software it don't really matter if you are running intel or m1?

**Reply by SMillerDev** [reply]
Brew doesn't care if it runs on Intel or M1. But most of the things brew installs cares very much about the cpu architecture. If you run the normal script without Rosetta and it works for you, that's the best case scenario.

If there is software that fails to install because you're on ARM you can always install a second instance  of brew with Rosetta.

**Reply by Sasatari** [reply]
I´m not very experienced with the shell, but long before Apple did, I changed my default shell from bash to zsh. That gave me some trouble with automated scripts that do a "do a shell script". When a program is linked in my (zsh-)Terminal, I just can call it anywhere (for example 'wine64'). But when doing it via Script Editor (to get an App-Button for programs), it fails and I have to start the program by writing down the whole path. I suspect (as said, I´m not very experienced) that script editor uses bash and the program, that I manually installed in my zsh-Terminal, isn´t linked there.

Long story, short question: As zsh now is the default shell, why do we use bash for brew install? Is it possible to simply say /bin/zsh -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)" ?

I´m asking, because I´ve just ordered a new MBP with M1 and want to make a clean install from the beginning, also create a tutorial for Mac-Users who want to run Altirra (Atari 8Bit-Emulator) on Big Sur. Installing Homebrew is the first step for this.

**Reply by SMillerDev** [reply]
That's really a separate question. But no, you can't run the installer in zsh. The installer uses some features of bash since all systems homebrew support have a version of bash. You can still run `bash` after you change your default she'll though so this isn't really affected AFAIK.
